libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1572,6 +1573,8 @@ void SMTEncoder::arrayPushPopAssign(Expression const& _expr, smtutil::Expression
 	else if (auto const* funCall = dynamic_cast<FunctionCall const*>(expr))
 	{
 		FunctionType const& funType = dynamic_cast<FunctionType const&>(*funCall->expression().annotation().type);
+		// Push cannot occur on an expression that is itself a ByteArrayPush, i.e., bytes.push().push() is not possible.
```

Comment:
How does the assertion guarantee that the expression here is a bytesarraypush?

Reply:
Not sure what you mean. It shouldn't be that